### ISSUE TITLE
Take scale into account when evaluating actor scene constraint

### ISSUE THREAD

**Issue by JohnSundell** [JohnSundell/ImagineEngine#65]
We should take an actor's scale into account when evaluating its `scene` constraint. Constraints enables users to restrict where and how actors can move, and an actor can be prevented from leaving its scene with the following code:

```swift
// In a scene
let actor = Actor()
actor.constraints = [.scene]
add(actor)
```

However, if a scale is set, the actor's original size will be used for the constraint calculations:
```
actor.size = Size(width: 100, height: 200)
actor.scale = 0.5
```

The actor will now still be treated as having a physical size of `100 x 200`, even though in reality it's `50 x 100`. Here's a screenshot from Revazendo that shows the problem:

![simulator screen shot - iphone 5s - 2017-11-11 at 13 01 53](https://user-images.githubusercontent.com/2466701/32689221-90f5d900-c6e0-11e7-9d14-12a2bb640e61.png)

**Good to know:**

- An actor's constraints are evaluated in the `Grid` class, in the `actorRectDidChange` method.
- An actor's `rect` is adjusted for its scale, so it should be used instead of its `size`.

**Comment by Juanpe** [comment]
Hi John, thanks for the iniciative! 👍🏼
I submitted a PR with a proposed change. Please let me know if I must make any further additions to solve this issue.

**Comment by JohnSundell** [comment]
Fixed by @Juanpe in #69! 🎉
